ha-example/README.md

```diff
@@ -0,0 +1,160 @@
+# Flink cluster HA configuration
+
+In the deployment of Flink cluster, it consists of operators, job managers, and task managers.
+By default, the deployment is not a high availability (HA) cluster, we need to configure it to achieve it.
+
+## Flink operator
+
+The Flink operator manages and operates Flink Deployments. It supports high availability through leader election and standby operator instances.
+To enable leader election we need to add the following two mandatory operator configuration parameters.
+
+```
+kubernetes.operator.leader-election.enabled: true
+kubernetes.operator.leader-election.lease-name: flink-operator-lease
+```
+The lease name must be unique in the current lease namespace.
+
+So, when installing flink operator using helm, we need to add these 2 configs + existing default configs via command like this:
+```
+helm install flink-kubernetes-operator flink-operator-repo/flink-kubernetes-operator \
+--set podSecurityContext=null \
+--set defaultConfiguration."log4j-operator\.properties"=monitorInterval\=30 \
+--set defaultConfiguration."log4j-console\.properties"=monitorInterval\=30 \
+--set replicas=2 \
+--set defaultConfiguration."flink-conf\.yaml"="kubernetes.operator.metrics.reporter.slf4j.factory.class\:\ org.apache.flink.metrics.slf4j.Slf4jReporterFactory
+kubernetes.operator.metrics.reporter.slf4j.interval\:\ 5\ MINUTE
+kubernetes.operator.reconcile.interval:\ 15\ s
+kubernetes.operator.observer.progress-check.interval:\ 5\ s
+kubernetes.operator.leader-election.enabled:\ true
+kubernetes.operator.leader-election.lease-name:\ flink-operator-lease" \
+-n flink
+```
+Note: The `replicas` config is to set the number of operator replicas.
+
+After running this command, you should see there are 2 operator replicas lanuched.
+```
+kubectl  get pod -n flink | grep flink-kubernetes-operator
+NAME                                          READY   STATUS    RESTARTS         AGE
+flink-kubernetes-operator-6cd86cc8-fmb2x      2/2     Running   0                3h
+flink-kubernetes-operator-6cd86cc8-g298v      2/2     Running   0                3h
+```
+
+And when checking the `lease` resource, we can see hte holder(leader) operator's name.
+```
+kubectl get lease -n flink
+NAME                       HOLDER                                      AGE
+flink-operator-lease       flink-kubernetes-operator-6cd86cc8-g298v    45h
+```
+
+When trying to delete the leader operator pod, we can see the holder will change to the other one (or the newly created one),
+which makes the operator highly available.
+
+## job manager
+
+The JobManager ensures consistency during recovery across TaskManagers. For the JobManager itself to recover consistently,
```

Comment:
If you make it one sentence per line then it is easier to comment. Also I suggest using soft not hard wrap in your editor, to prevent lines getting split.